Extension/src/LanguageServer/client.ts

```diff
@@ -868,6 +877,77 @@ export class DefaultClient implements Client {
         return workspaceFolder ? workspaceFolder.name : "untitled";
     }
 
+    public updateClientConfigurations(): void {
+        clients.forEach(client => {
+            if (client instanceof DefaultClient) {
+                const defaultClient: DefaultClient = <DefaultClient>client;
+                defaultClient.configuration.CompilerDefaults = compilerDefaults;
+                defaultClient.configuration.handleConfigurationChange();
+            }
+        });
+    }
+
+    public async handleCompilerQuickPick(): Promise<void> {
+        const compilerPath: PersistentState<string[]> = new PersistentState<string[]>("CPP.trustedCompilerPaths", []);
+        const paths: string[] = compilerDefaults.knownCompilers.map(function (a: configs.KnownCompiler): string { return a.path; });
+        paths.push("Select another compiler on my machine");
+        paths.push("Help me install a compiler");
+        paths.push("Do not configure a compiler (not recommended)");
+        const index: number = await ui.showSelectCompiler(paths);
+        util.updateTrustedCompilersList(paths[index]);
```

Comment:
It looks like this is adding `path[index]` to the list of trusted compilers before checking if the user selected one of the compiler entries.  Should this maybe go into a final else?

Also, the function name is a bit vague.  Maybe `addTrustedCompiler` instead of `updateTrustedCompilersList` as that more accurately reflects what it does?